app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -451,6 +453,14 @@ class WireActivityViewModel @Inject constructor(
 
                     is ObservePersistentWebSocketConnectionStatusUseCase.Result.Success -> {
                         result.persistentWebSocketStatusListFlow.collect { statuses ->
+                            val usersToObserve = statuses
+                                .filter { !it.isPersistentWebSocketEnabled }
+                                .map { it.userId }
+
+                            notificationManager.observeNotificationsAndCallsWhileRunning(
```

Comment:
If the delay added in `handleConnectionOnPushNotification` makes it work then is this move from `GlobalObserversManager` to `WireActivityViewModel` necessary? I mean, now we have a single activity so maybe it's safe to assume "WireActivity is open == the app is running" but that can not always be the case and I don't quite understand how moving it here is a solution, could you explain why we need to move it here? 🤔 Also, won't that create a race condition with creating the notification channels because both are executed independently now?

Reply:
moving it here doesn't actually solve any known issue. But the problem of having it in `GlobalObserversManager`  is a coroutine scope we use for observation. 
In `GlobalObserversManager` we use scope that lives longer, than ViewModelScope so that we may observe the messages longer than the app lives, which is redundant. 
Once the app is hidden/killed we disconnect from the websocket and if user receives anything, we'll get FCM push which triggers a new connection and new messages observation.

Reply:
about independent notifications and channels creating: you're right, it's rare, but possible case, so I'll move it to ViewModel too.

Reply:
> we may observe the messages longer than the app lives

You mean longer than the Activity lives, right? Because `GlobalObserversManager` is also a part of the app so if it lives, it means that the app also still lives. 😅 On the other hand, we can have gaps between the Activity is being closed, websocket being closed and the app starts receiving pushes and maybe handling notification twice in that case is still better than missing the notification until next one arrives 😅 Maybe I'm wrong but redundancy still shouldn't happen because in `observeMessageNotificationsOnceJob` we check if `observingWhileRunningJobs` is still running 🤔 

> about independent notifications and channels creating: you're right, it's rare, but possible case, so I'll move it to ViewModel too.

I may be dangerous, I think we already had that in the ViewModel and it was causing issues, because channels were created too late - for instance when your app updates in the background and you don't open the app but receive a push - until you open the app and create this ViewModel you don't have channels created. 🙈

Reply:
Yeah, you are right about ViewModel. 
So I moved the channelCreating and notificationObserving back to the GlobalObserversManager AND added there running `PersistentWebSocketService` so we don't do `observeNotificationsAndCallsWhileRunning` for the users that have `PersistentWebSocket` ON  :)